presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java

```diff
@@ -0,0 +1,153 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.airlift.tpch.LineItem;
+import io.airlift.tpch.LineItemGenerator;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createUnboundedVarcharType;
+import static com.google.common.util.concurrent.MoreExecutors.listeningDecorator;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@State(Scope.Thread)
+@OutputTimeUnit(SECONDS)
+@Fork(3)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+public class BenchmarkBinaryFileSpiller
+{
+    private static final BlockEncodingSerde BLOCK_ENCODING_MANAGER = new BlockEncodingManager(new TypeRegistry(ImmutableSet.of(BIGINT, DOUBLE, VARCHAR)));
+
+    @Benchmark
+    public void write(BenchmarkData data)
+            throws ExecutionException, InterruptedException
+    {
+        try (Spiller spiller = new BinaryFileSpiller(BLOCK_ENCODING_MANAGER, data.getExecutor())) {
+            spiller.spill(data.getPages().iterator()).get();
+        }
+    }
+
+    @Benchmark
+    public void read(BenchmarkData data)
+    {
+        List<Iterator<Page>> spills = data.getSpiller().getSpills();
+        for (Iterator<Page> spill : spills) {
+            while (spill.hasNext()) {
+                Page next = spill.next();
+                next.getPositionCount();
+            }
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private ListeningExecutorService executor;
+
+        @Param({"10000"})
+        private int rowsPerPage = 10000;
+
+        @Param({"10"})
+        private int pagesCount = 10;
+
+        private List<Page> pages;
+        private Spiller spiller;
+
+        @Setup
+        public void setup()
+                throws ExecutionException, InterruptedException
+        {
+            executor = listeningDecorator(newSingleThreadScheduledExecutor());
```

Comment:
I neither changed benchmark results nor improved stability but I have replaced it. It might be easier to debug. On the other hand it makes this benchmark a little bit more artificial.